tareas/Tarea_2_Andres/intervalos.jl

```diff
@@ -0,0 +1,352 @@
+module Intervalos
+
+export Intervalo, intervalo_vacio, isinterior, ⪽, hull, ⊔, division_extendida
+
+struct Intervalo{T<:Real}
+	
+	infimo::T;   supremo::T
+	
+# To promote every subtype of Real to AbstractFloat, and then promote them to the one with more hierarchy, like Float64 or BigFloat
+	function Intervalo(infimo::T1, supremo::T2) where {T1<:Real, T2<:Real}
+		inf, sup, _ = promote(infimo, supremo, 1.0)
+		return Intervalo(inf, sup)
+	end
+
+# To check that the interval satisfy the condition a ≤ b for Intervalo(a,b) ≡ [a,b]	
+	function Intervalo(infimo::T, supremo::T) where {T<:AbstractFloat}
+		if isnan(infimo) || isnan(supremo)
+			return new{T}(NaN, NaN)
+		else
+			@assert infimo ≤ supremo
+			return new{T}(infimo, supremo)
+		end
+	end
+
+# To allow that Intervalo(a) ≡ [a] → [a, a] also work
+	function Intervalo(same::T) where {T<:Real}
+		return Intervalo(same, same)
+	end
+end
+
+begin   ### Empty interval
```

Comment:
```suggestion
"""
   intervalo_vacio()  
   
   Función que devuelve un intervalo sin recibir algún argumento.
"""
```

Comienza a eliminar begins y ends, y agregar a cada función una descripción acerca de esta. Esto incluye los argumentos que recibe, que hace la función e incluso, si es necesario que retorna esta función.